Add REST API endpoint for querying configuration settings

buchi opened the pull request:
Provides a new endpoint for querying configuration settings that may be needed in clients.

Currently it returns:
- maximum subdossier depth
- maximum repository depth
- activated and disabled features

Example Response:
```json
{
    "@id": "http://localhost:8080/fd/@config",
    "features": {
        "activity": true,
        "bumblebee": false,
        "contacts": false,
        "dossier_templates": true,
        "meetings": true,
        "officatwork": true,
        "solr": true,
        "workspace": false
    },
    "maximum_repository_depth": 3,
    "maximum_subdossier_depth": 1
}
```

jone commented:
There is also a list of features here: https://github.com/4teamwork/opengever.core/blob/master/opengever/testing/integration_test_case.py#L37
Wouldn't it make sense to extract both into one common list?
I also think that we have more features in the tests than here at the moment 😉

lukasgraf commented on opengever/api/config.py:
```diff
@@ -0,0 +1,47 @@
+            'is_feature_enabled', interface=IDossierTemplateSettings)
+        config['features']['meetings'] = api.portal.get_registry_record(
+            'is_feature_enabled', interface=IMeetingSettings)
+        config['features']['officatwork'] = api.portal.get_registry_record(
```
```
officatwork  -> officeatwork
                     ^
```

buchi replied:
Fixed.

lukasgraf commented on opengever/api/config.py:
```diff
@@ -0,0 +1,47 @@
+        config['maximum_repository_depth'] = api.portal.get_registry_record(
+            'maximum_repository_depth', interface=IRepositoryFolderRecords)
+
+        config['maximum_subdossier_depth'] = api.portal.get_registry_record(
```
As discussed, I would use this opportunity to use constistent semantics for these nesting depth settings (`maximum_subdossier_depth` refers to only *sub*dossiers, whereas `maximum_repository_depth` refers to nesting of repository folders inside themselves.

I would therefore simply call them
- `max_dossier_levels`
- `max_repositoryfolder_levels`

and adjust on of them by one accordingly compared to the internally stored value.

(I find `_level` easier to visualize, personally).

The meaning then is:
`1` - no nesting allowed, so no sub-dossiers or sub-repofolders
`2` - exactly one sub-object allowed
etc...

@deiferni thoughts?

deiferni replied:
As discussed i agree, i also find levels easier to understand 👍 . Also since this is read-only we don't have to bother with preventing someone from setting `0` as the value yet.

buchi replied:
changed to `max_repositoryfolder_levels` and `max_dossier_levels`.

lukasgraf commented on docs/HISTORY.txt:
```diff
@@ -8,6 +8,7 @@ Changelog
 - Remove all uses of enableHTTPCompression in GEVER. [lgraf]
 - Fix deleting agenda items for non-word agenda items. [deiferni]
 - Improve checking for truthy dates in sablon template data. [deiferni]
+- Add REST API endpoint for querying configuration settings. [buchi]
```
Documentation is missing for https://docs.onegovgever.ch/dev-manual/api/

lukasgraf commented:
@jone I agree. I'm also less than happy that we're duplicating that logic here, and inconsistently so. But apparently this is urgent.

So I see this as a chance to commit to a "public" configuration schema (a small subset of what's currently configurable, but still). Once we publish this, and clients rely on it, we can't change it in non-backwards compatible ways.

Therefore we can consider this the canonical source of truth for naming settings in the future, eventually refactor names in our internal codebase, and use this to inform names in other contexts where we need to specify settings (OGGBundle import, policies, OGGBundle Validator, Excel sheets for Onboarding of new customers, tests)

deiferni commented on opengever/api/config.py:
```diff
@@ -0,0 +1,47 @@
+        config['features'] = {}
+        config['features']['activity'] = api.portal.get_registry_record(
+            'is_feature_enabled', interface=IActivitySettings)
+        config['features']['bumblebee'] = api.portal.get_registry_record(
```
As already suggested in a discussion by @lukasgraf i'd not call this `bumblebee` but rather `pdf_preview`, or maybe just `preview`. The name `bumblebee` is internal only and should not be exposed to customers/API clients.

buchi replied:
Renamed to `preview`.

buchi commented:
I've added some more features found in https://github.com/4teamwork/opengever.core/blob/master/opengever/testing/integration_test_case.py#L37

Also added docs for `@config` endpoint.
